File: src/main/java/com/chatdemo/config/FirebaseConfig.java
package com.chatdemo.config;

/**
 * Configuration for Firebase Storage uploads.
 */
public final class FirebaseConfig {

    private static final String DEFAULT_SERVICE_ACCOUNT_PATH = "firebase_settings.json";

    private FirebaseConfig() {
    }

    public static String getServiceAccountPath() {
        String path = System.getenv("FIREBASE_SERVICE_ACCOUNT_PATH");
        if (path != null && !path.isBlank()) {
            return path;
        }
        return DEFAULT_SERVICE_ACCOUNT_PATH;
    }

    public static String getStorageBucket() {
        String bucket = System.getenv("FIREBASE_STORAGE_BUCKET");
        if (bucket != null && !bucket.isBlank()) {
            return bucket;
        }
        return null;
    }
}

File: src/main/scala/com/chatdemo/backend/config/FirebaseConfig.scala
package com.chatdemo.backend.config

/**
 * Configuration for Firebase Storage uploads.
 */
object FirebaseConfig {

  private val DefaultServiceAccountPath = "firebase_settings.json"

  def getServiceAccountPath: String = {
    val path = System.getenv("FIREBASE_SERVICE_ACCOUNT_PATH")
    if (path != null && !path.isBlank) {
      path
    } else {
      DefaultServiceAccountPath
    }
  }

  def getStorageBucket: String = {
    val bucket = System.getenv("FIREBASE_STORAGE_BUCKET")
    if (bucket != null && !bucket.isBlank) {
      bucket
    } else {
      null
    }
  }
}
